WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -281,25 +270,49 @@ bool TryRemoveTrackedAutoStart(IWallet wallet)
 
 		await WaitAndHandleResultOfTasksAsync(nameof(trackedAutoStarts), trackedAutoStarts.Values.Select(x => x.Task).ToArray()).ConfigureAwait(false);
 	}
-
+	
+	private bool TryRemoveTrackedAutoStart(ConcurrentDictionary<IWallet, TrackedAutoStart> trackedAutoStarts, IWallet wallet)
+	{
+		if (trackedAutoStarts.TryRemove(wallet, out var trackedAutoStart))
+		{
+			trackedAutoStart.CancellationTokenSource.Cancel();
+			trackedAutoStart.CancellationTokenSource.Dispose();
+			return true;
+		}
+		return false;
+	}
+	
 	private void ScheduleRestartAutomatically(IWallet walletToStart, ConcurrentDictionary<IWallet, TrackedAutoStart> trackedAutoStarts, bool stopWhenAllMixed, bool overridePlebStop, CancellationToken stoppingToken)
 	{
+		var skipDelay = false;
 		if (trackedAutoStarts.ContainsKey(walletToStart))
 		{
-			walletToStart.LogDebug("AutoStart was already scheduled.");
-			return;
+			if (stopWhenAllMixed == trackedAutoStarts[walletToStart].StopWhenAllMixed && overridePlebStop == trackedAutoStarts[walletToStart].OverridePlebStop)
+			{
+				walletToStart.LogDebug("AutoStart was already scheduled");
+				return;
+			}
+			
+			walletToStart.LogDebug("AutoStart was already scheduled with different parameters, cancel the last task and do not wait.");
+			TryRemoveTrackedAutoStart(trackedAutoStarts, walletToStart);
+			skipDelay = true;
 		}
+		
+		NotifyWalletStartedCoinJoin(walletToStart);
 
 #pragma warning disable CA2000 // Dispose objects before losing scope
 		var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(stoppingToken);
 #pragma warning restore CA2000 // Dispose objects before losing scope
 
-		var restartTask = Task.Run(
+		var restartTask = new Task(
```

Comment:
Why the change here?

Reply:
Sorry I did not see that. How can concurrency happen? 

https://github.com/molnard/WalletWasabi/blob/fafc8bfa/WalletWasabi/WabiSabi/Client/CoinJoinManager.cs#L256